src/sentry/lang/native/sources.py

```diff
@@ -577,6 +580,30 @@ def resolve_alias(source):
         if key not in builtin_sources:
             continue
 
+        if features.has("organizations:gcp-bearer-token-authentication", organization):
+            if source.get("type") == "gcs":
+                # Fetch the regular credentials for GCP
+                source_credentials, _ = google.auth.default()
+
+                # Impersonate the service account to give the token for symbolicator a proper scope
+                target_credentials = impersonated_credentials.Credentials(
+                    source_credentials=source_credentials,
+                    target_principal=source.get("client_email"),
+                    target_scopes=["https://www.googleapis.com/auth/cloud-platform"],
+                    lifetime=3600,
+                )
+
+                target_credentials.refresh(Request())
+
+                if target_credentials.token is not None:
+                    source["bearer_token"] = target_credentials.token
+
+                    # Remove other credentials if we have a token
+                    if "client_email" in source:
+                        del source["client_email"]
+                    if "private_key" in source:
+                        del source["private_key"]
```

Comment:
I don't think you need these conditionals here and can unconditionally delete the keys.

Reply:
Trying to delete a nonexistent key from a dict throws an exception. Or do you mean because they must definitely be present because of the schema?

Reply:
Ah yeah true, `source.pop("client_email", None)` works, not sure if that's better.